feat: add `onError` event handler

lihbr opened the pull request:
<!--- Provide a general summary of your changes in the title above -->

## Types of changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->

- [ ] Chore (a non-breaking change which is related to package maintenance)
- [ ] Bug fix (a non-breaking change which fixes an issue)
- [x] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Description

<!--- Describe your changes in detail -->
Add a minimal `onError` event handler to `createRPCMiddleware` and `handleRPCRequest`
<!--- Why is this change required? What problem does it solve? -->
<!--- If it resolves an open issue, please link to the issue here. For example "Resolves: #137" -->

## Checklist:

<!--- Put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

- [x] My change requires an update to the official documentation.
- [x] All [TSDoc](https://tsdoc.org) comments are up-to-date and new ones have been added where necessary.
- [x] All new and existing tests are passing.

<!--- A cute animal (or car) picture is welcome to close your PR! -->

lihbr committed:
feat: add `onError` event handler

lihbr committed:
docs: document `onError` event handler

lihbr committed:
build: release as path

lihbr committed:
build: release as patch

lihbr committed:
chore(release): 0.1.7-alpha.0

lihbr committed:
docs: code style

angeloashmore committed:
feat: wrap r19-specific errors with `R19Error`

angeloashmore committed:
feat: export `R19Error` and `isR19ErrorLike` helper

angeloashmore commented:
Looks good! I opened #9 to make the solution a bit more complete.

We can iterate on that PR based on your review. Otherwise, we're good to merge.

angeloashmore committed:
fix: don't wrap procedure-thrown errors

angeloashmore committed:
fix: remove `isR19ErrorLike()`

The helper is not needed since we re-throw the error on the client as an `R19Error`.

lihbr committed:
Merge pull request #9 from prismicio-community/feat/add-on-error-event-handler-enhancements

feat: wrap r19-specific errors with `R19Error`
